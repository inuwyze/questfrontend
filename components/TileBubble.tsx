import React, { PropsWithChildren, ReactNode } from "react"
import { View, Text, StyleSheet } from "react-native"
import { ITileBubbleProp } from "../types/QuestTypes"

const TileBubble:React.FC<PropsWithChildren<ReactNode>>=({children})=>{
    return(
        <View
        style={styles.container}>
            {children}
        </View>
    )
}

const styles=StyleSheet.create({
    container:{    
        
        borderColor:'black',
        borderBottomWidth:4,
        borderRightWidth:4,
        padding:20,
        borderRadius:10,
        marginVertical:10,
        elevation:10,
       
        backgroundColor:'white'
    },

})

export default TileBubble