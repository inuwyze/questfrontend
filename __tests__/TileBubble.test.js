import React from 'react';
import {Text} from 'react-native'
import renderer from 'react-test-renderer';
import TileBubble from '../components/TileBubble';



describe('Testing TileBubble', () => {
  it('has null child when nothing is passed', () => {
    const tree = renderer.create(<TileBubble />).toJSON();
    // console.log(tree)
    expect(tree.children).toBe(null);
  });
  it('has child when nothing is passed', () => {
    const tree = renderer.create(<TileBubble >
        <Text>hello</Text>
        </TileBubble>
        ).toJSON();
    // console.log(tree)
    expect(tree.children.length).toBe(1);
  });
  it('Tile Bubble renders correctly', () => {
    const tree = renderer.create(<TileBubble />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});