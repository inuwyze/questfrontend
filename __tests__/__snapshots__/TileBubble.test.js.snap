// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TileBubble Tile Bubble renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderBottomWidth": 4,
      "borderColor": "black",
      "borderRadius": 10,
      "borderRightWidth": 4,
      "elevation": 10,
      "marginVertical": 10,
      "padding": 20,
    }
  }
/>
`;
